homeassistant/components/switch/mystrom.py

```diff
@@ -0,0 +1,104 @@
+"""
+homeassistant.components.switch.mystrom
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Support for myStrom switches.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/switch.mystrom/
+"""
+import logging
+import requests
+
+from homeassistant.components.switch import SwitchDevice
+
+DEFAULT_NAME = 'myStrom Switch'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument, too-many-function-args
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Find and return myStrom switches. """
+    resource = config.get('resource')
```

Comment:
I think host would be more appropriate to use here, as that is what is used as the base for all the urls.